### PR TITLE
Fix the --loglevel flag

### LINKED ISSUE #17: Add verbose option for output
This is the current output that I think might be good to move (at least majority of it) to a `--verbose` option. This will help minimize the output and most users wouldn't change all this info:
```
Using machine config: frontier.ini
Overriding tasks in inputfile since CLI mode was provided
runmodecmd =  ['start', 'stop']
self.__harness_task:  [['start_tests', None, None]]
self.__harness_task:  [['start_tests', None, None], ['stop_tests', None, None]]
reading harness config /sw/acceptance/olcf-test-harness/configs/frontier.ini
RGT_PATH_TO_SSPACE is already set. Skipping.
App_test_list: [<libraries.apptest.subtest object at 0x7fa807de4748>]
Tasks: [['start_tests', None, None], ['stop_tests', None, None]]
Using machine config: frontier.ini
reading harness config /sw/acceptance/olcf-test-harness/configs/frontier.ini
RGT_MACHINE_NAME is already set. Skipping.
RGT_MACHINE_TYPE is already set. Skipping.
RGT_SCHEDULER_TYPE is already set. Skipping.
RGT_JOBLAUNCHER_TYPE is already set. Skipping.
RGT_CPUS_PER_NODE is already set. Skipping.
RGT_GPUS_PER_NODE is already set. Skipping.
RGT_SUBMIT_QUEUE is already set. Skipping.
RGT_SUBMIT_ARGS is already set. Skipping.
RGT_NCCS_TEST_HARNESS_MODULE is already set. Skipping.
RGT_TYPE_OF_REPOSITORY is already set. Skipping.
RGT_GIT_REPS_BRANCH is already set. Skipping.
RGT_GIT_DATA_TRANSFER_PROTOCOL is already set. Skipping.
RGT_GIT_MACHINE_NAME is already set. Skipping.
RGT_GIT_SERVER_APPLICATION_PARENT_DIR is already set. Skipping.
RGT_GIT_SSH_SERVER_URL is already set. Skipping.
RGT_GIT_HTTPS_SERVER_URL is already set. Skipping.
RGT_PATH_TO_SSPACE is already set. Skipping.
RGT_SYSTEM_LOG_TAG is already set. Skipping.
RGT_ACCT_ID is already set. Skipping.
Generated test unique id: 1636033386.3722255
Creating machine frontier: Type = linux_x86_64 ; Scheduler = slurm ; Job launcher = srun
Creating scheduler
Creating jobLauncher
```

### LINKED ISSUE #37: No error message on the console when missing job template
When a test is missing the job template, like slurm.template.x, there is no warning output. The submit exit code is still listed as 0, and the Run_Archive and workdir directories are completely empty, and the only indication that no job template was found is in LogFiles:
```
-----
Time: 2022-07-14 15:42:05,387
Logger: __main__
Loglevel: CRITICAL
Message:
In function make_batch_script_for_linux: Error opening bath template file 'slurm.template.x' for reading.Handling error: [Errno 2] No such file or directory: 'slurm.template.x'
-----
```
Some message should be logged to the console in this event.

### LINKED ISSUE #102: LogLevel flags hard-coded in several object classes
There are several classes that have hard-coded loglevel/threshold levels. For example, in `harness/libraries/regresstion_test.py`:
```
                logger_threshold = self.__log_level
                fh_threshold_log_level = "INFO"
                ch_threshold_log_level = "CRITICAL"
                a_logger = rgt_logger_factory.create_rgt_logger(logger_name=logger_name,
                                      fh_filepath=fh_filepath,
                                      logger_threshold_log_level=logger_threshold,
                                      fh_threshold_log_level=fh_threshold_log_level,
                                      ch_threshold_log_level=ch_threshold_log_level)
```
The logger threshold is inherited, which is good, but the file handler & console handler log levels are hardcoded.

### PR DESCRIPTION
**Pull request by hagertnl**
Currently, the --loglevel flag does not affect verbosity on the command line/stdout. I would like to change that so that it does affect verbosity on the command line.

### CODE REVIEW

**Comment by hagertnl** [comment]
A quick note on the logging module and the log levels:
The logger_threshold_level filters *all* log messages before they ever reach the file or console handlers.
So if you set logger_threshold_level to WARNING, then the file handler will only receive messages >= WARNING.
Then the fh_threshold_level and ch_threshold_level decide what level of messages to send to the file and console, respectively.

Major changes:
- Added a `--loglevel` flag to `test_harness_driver.py`
- Changed the `--loglevel` functionality so that instead of `--loglevel` changing the `logger_threshold_level`, it only changes the console log level.
    - the filehandler (log file) is hard-coded to INFO, and the `logger_threshold_level` is either set to `NOTSET` or `DEBUG`.

**Comment by hagertnl** [comment]
Tested, and I think that this PR will need to also include a re-assigning of log levels to messages. There are some messages that aren't assigned proper priority. For example, here is the loglevel set to CRITICAL:
```
> runtests.py -i rgt.input.unittests --mode start stop --loglevel CRITICAL
Using machine config: crusher.ini
Using machine config: crusher.ini
Overriding tasks in inputfile since CLI mode was provided
runmodecmd =  ['start', 'stop']
self.__harness_task:  [['start_tests', None, None]]
self.__harness_task:  [['start_tests', None, None], ['stop_tests', None, None]]
reading harness config /lustre/orion/proj-shared/stf016/hagertnl/Scratch/harness-dev/olcf-test-harness/configs/crusher.ini
RGT_PATH_TO_SSPACE is already set. Skipping.
Starting tasks for Application.Test: harness_unit_tests.test_successful_run: [['start_tests', None, None], ['stop_tests', None, None]]
Using machine config: crusher.ini
reading harness config /lustre/orion/proj-shared/stf016/hagertnl/Scratch/harness-dev/olcf-test-harness/configs/crusher.ini
RGT_MACHINE_NAME is already set. Skipping.
RGT_MACHINE_TYPE is already set. Skipping.
RGT_SCHEDULER_TYPE is already set. Skipping.
RGT_JOBLAUNCHER_TYPE is already set. Skipping.
RGT_CPUS_PER_NODE is already set. Skipping.
RGT_GPUS_PER_NODE is already set. Skipping.
RGT_SUBMIT_QUEUE is already set. Skipping.
RGT_SUBMIT_ARGS is already set. Skipping.
RGT_NCCS_TEST_HARNESS_MODULE is already set. Skipping.
RGT_TYPE_OF_REPOSITORY is already set. Skipping.
RGT_GIT_REPS_BRANCH is already set. Skipping.
RGT_GIT_DATA_TRANSFER_PROTOCOL is already set. Skipping.
RGT_GIT_MACHINE_NAME is already set. Skipping.
RGT_GIT_SERVER_APPLICATION_PARENT_DIR is already set. Skipping.
RGT_GIT_SSH_SERVER_URL is already set. Skipping.
RGT_GIT_HTTPS_SERVER_URL is already set. Skipping.
RGT_PATH_TO_SSPACE is already set. Skipping.
RGT_SYSTEM_LOG_TAG is already set. Skipping.
RGT_ACCT_ID is already set. Skipping.
Using launch id: crusher_test/hagertnl@2023-05-12T09:18:56.59
Generated test unique id: 1683897536.7776563
In test_harness_driver, creating logger with logger_name=__main__, fh_filepath=/autofs/nccs-svm1_proj/stf016/hagertnl/harness/unit_tests/harness_unit_tests/test_successful_run/Run_Archive/1683897536.7776563/LogFiles/application_logfile.txt
Creating machine crusher: Type = linux_x86_64 ; Scheduler = slurm ; Job launcher = srun
Creating scheduler
Creating jobLauncher
RGT Test Parameters - Builtin
=============================
job_name = test_successful_run
nodes = 1
total_processes = 1
batch_queue = batch
walltime = 1
executable_path = ./dummy
batch_filename = run.sh
build_cmd = ./build_success.sh
check_cmd = ./check.sh
report_cmd = ./report.sh
resubmit = 1
RGT Test Parameters - User
==========================
Path to Source: /autofs/nccs-svm1_proj/stf016/hagertnl/harness/unit_tests/harness_unit_tests/Source
Path to Build Dir: /lustre/orion/proj-shared/stf016/hagertnl/harness_sspace/crusher/04.27.23-16.26/harness_unit_tests/test_successful_run/1683897536.7776563/build_directory
WARNING: Ignoring invalid built-in parameter key launch_id
Submitting job from SLURM class using batchfilename run.sh
sbatch  -p batch  -A stf016 run.sh
SLURM jobID =  313792
build exit value = 0
submit exit value = 0
Skipped 0, launched 1.
```

**Comment by hagertnl** [comment]
After the most recent updates, below is what we see when using CRITICAL log threshold. One note. The `build exit value` and `submit exit value` lines typically at the bottom of the output are no longer there, because I added them as INFO logging threshold, instead of `print()`. To my knowledge, there's not a logger threshold that is "critical info" - info that should be printed every time. Should I use the CRITICAL threshold for this?
```
> runtests.py -i rgt.input.unittests --mode start stop --loglevel CRITICAL
Using machine config: crusher.ini
Using machine config: crusher.ini
Overriding tasks in inputfile since CLI mode was provided
runmodecmd =  ['start', 'stop']
self.__harness_task:  [['start_tests', None, None]]
self.__harness_task:  [['start_tests', None, None], ['stop_tests', None, None]]
reading harness config /lustre/orion/proj-shared/stf016/hagertnl/Scratch/harness-dev/olcf-test-harness-fork/configs/crusher.ini
RGT_PATH_TO_SSPACE is already set. Skipping.
Starting tasks for Application.Test: harness_unit_tests.test_successful_run: [['start_tests', None, None], ['stop_tests', None, None]]
Using machine config: crusher.ini
reading harness config /lustre/orion/proj-shared/stf016/hagertnl/Scratch/harness-dev/olcf-test-harness-fork/configs/crusher.ini
RGT_MACHINE_NAME is already set. Skipping.
RGT_MACHINE_TYPE is already set. Skipping.
RGT_SCHEDULER_TYPE is already set. Skipping.
RGT_JOBLAUNCHER_TYPE is already set. Skipping.
RGT_CPUS_PER_NODE is already set. Skipping.
RGT_GPUS_PER_NODE is already set. Skipping.
RGT_SUBMIT_QUEUE is already set. Skipping.
RGT_SUBMIT_ARGS is already set. Skipping.
RGT_NCCS_TEST_HARNESS_MODULE is already set. Skipping.
RGT_TYPE_OF_REPOSITORY is already set. Skipping.
RGT_GIT_REPS_BRANCH is already set. Skipping.
RGT_GIT_DATA_TRANSFER_PROTOCOL is already set. Skipping.
RGT_GIT_MACHINE_NAME is already set. Skipping.
RGT_GIT_SERVER_APPLICATION_PARENT_DIR is already set. Skipping.
RGT_GIT_SSH_SERVER_URL is already set. Skipping.
RGT_GIT_HTTPS_SERVER_URL is already set. Skipping.
RGT_PATH_TO_SSPACE is already set. Skipping.
RGT_SYSTEM_LOG_TAG is already set. Skipping.
RGT_ACCT_ID is already set. Skipping.
Using launch id: crusher_test/hagertnl@2023-08-29T09:54:13.79
Generated test unique id: 1693317253.9446626
Creating machine crusher: Type = linux_x86_64 ; Scheduler = slurm ; Job launcher = srun
Creating scheduler
Creating jobLauncher
RGT Test Parameters - Builtin
=============================
job_name = test_successful_run
nodes = 1
total_processes = 1
batch_queue = batch
walltime = 1
executable_path = ./dummy
batch_filename = run.sh
build_cmd = ./build_success.sh
check_cmd = ./check.sh
report_cmd = ./report.sh
resubmit = 1
RGT Test Parameters - User
==========================
WARNING: Ignoring invalid built-in parameter key launch_id
Submitting job from SLURM class using batchfilename run.sh
sbatch  -p batch  -A stf016 run.sh
SLURM jobID =  366639
Launched 1 tests, skipped 0 tests.
```

**Comment by hagertnl** [comment]
Most of the output currently is from files without loggers, which I identified in #130

File: harness/libraries/repositories/single_app_git_repository.py [hunk 1]
```diff
@@ -133,11 +133,11 @@ def cloneRepository(self,
         elif clone_flag == GitCloneFlag.FOUND_EXISTING_REPOSITORY_WITH_CORRECT_ORIGIN:
             message = "The directory {} exists and is already cloned. Therefore we will will skip cloning repository {}.\n".format(pathspec,
                                                                                                                                    self.remote_repository_URL)
-            logger.doInfoLogging(message)
+            logger.doWarningLogging(message)
         elif clone_flag == GitCloneFlag.FOUND_EXISTING_REPOSITORY_WITH_INCORRECT_ORIGIN:
             message = "The directory {} is an existing git repository whose origin is not {}.\n".format(pathspec,
                                                                                                         self.remote_repository_URL)
-            logger.doInfoLogging(message)
+            logger.doWarningLogging(message)
```
Comment (ddietz89) [inline harness/libraries/repositories/single_app_git_repository.py]: Should this be an error since it is raising an exception?

**Review by ddietz89 — CHANGES_REQUESTED** [review]
I think this looks good, other than the one question.

File: harness/libraries/repositories/single_app_git_repository.py (hunk 1 above)
Comment (hagertnl) [reply harness/libraries/repositories/single_app_git_repository.py]: Fixed! Made Critical, since it's immediately before an exception is thrown.

**Review by ddietz89 — APPROVED** [review]

**Pull request merged**